pkg/pool-stable-phantom/test/StablePoolStorage.test.ts

```diff
@@ -0,0 +1,330 @@
+    let exemptFromYieldProtocolFeeFlags: boolean[] = [];
+
+    async function deployPool(
+      tokens: TokenList,
+      numRateProviders = tokens.length,
+      numExemptFlags = tokens.length
+    ): Promise<void> {
+      const newRateProviders = [];
+      for (let i = 0; i < numRateProviders; i++) {
+        newRateProviders[i] = (await deploy('v2-pool-utils/MockRateProvider')).address;
+      }
+
+      const newExemptFromYieldProtocolFeeFlags = [];
+      for (let i = 0; i < numExemptFlags; i++) {
+        newExemptFromYieldProtocolFeeFlags[i] = i % 2 == 0; // set true for even tokens
+      }
+
+      pool = await deploy('MockStablePoolStorage', {
+        args: [vault.address, tokens.addresses, newRateProviders, newExemptFromYieldProtocolFeeFlags],
+      });
+      bptIndex = (await pool.getBptIndex()).toNumber();
+      rateProviders = newRateProviders;
+      exemptFromYieldProtocolFeeFlags = newExemptFromYieldProtocolFeeFlags;
+    }
+
+    sharedBeforeEach('deploy pool', async () => {
+      await deployPool(tokens);
+    });
+
+    describe('constructor', () => {
+      context('when the constructor succeeds', () => {
+        it('sets BPT index correctly', async () => {
+          const bpt = await Token.deployedAt(pool);
+          const allTokens = new TokenList([...tokens.tokens, bpt]).sort();
+          const expectedIndex = allTokens.indexOf(bpt);
+          expect(await pool.getBptIndex()).to.be.equal(expectedIndex);
+        });
+
+        it('sets the tokens', async () => {
+          const bpt = await Token.deployedAt(pool);
+          const allTokens = new TokenList([...tokens.tokens, bpt]).sort();
+
+          const expectedTokenAddresses = Array.from({ length: 6 }, (_, i) => allTokens.addresses[i] ?? ZERO_ADDRESS);
+          expect(await pool.getToken0()).to.be.eq(expectedTokenAddresses[0]);
+          expect(await pool.getToken1()).to.be.eq(expectedTokenAddresses[1]);
+          expect(await pool.getToken2()).to.be.eq(expectedTokenAddresses[2]);
+          expect(await pool.getToken3()).to.be.eq(expectedTokenAddresses[3]);
+          expect(await pool.getToken4()).to.be.eq(expectedTokenAddresses[4]);
+          expect(await pool.getToken5()).to.be.eq(expectedTokenAddresses[5]);
+        });
+
+        it('sets the fee exemption flags correctly', async () => {
+          const bpt = await Token.deployedAt(pool);
+          const allTokens = new TokenList([...tokens.tokens, bpt]).sort();
+
+          const expectedExemptFromYieldProtocolFeeFlags = exemptFromYieldProtocolFeeFlags.slice();
+          expectedExemptFromYieldProtocolFeeFlags.splice(bptIndex, 0, false);
+
+          for (let i = 0; i < allTokens.length; i++) {
+            // Initialized to true for even tokens
+            const expectedFlag = expectedExemptFromYieldProtocolFeeFlags[i];
+            const token = allTokens.get(i);
+
+            expect(await pool.isTokenExemptFromYieldProtocolFee(token.address)).to.equal(expectedFlag);
```

Comment:
We have two flavors of this, one that receives a token address and one that receives an index. It'd be good to do both.